drone_localization/core/services/retrieval.py

```
@@ -1,308 +1,101 @@
-
-        return gallery_embeddings, paths_list
+    # Вспомогательные методы для конвертации base64 <-> PIL
+    @staticmethod
+    def _b64_to_pil(b64_str: str) -> Image.Image:
+        if "base64," in b64_str:
+            b64_str = b64_str.split("base64,")[1]
+        img_bytes = base64.b64decode(b64_str)
+        return Image.open(io.BytesIO(img_bytes)).convert("RGB")
+
+    @staticmethod
+    def _pil_to_b64(img: Image.Image, format="JPEG") -> str:
```

Ruff (0.15.12)

[error] Function argument `format` is shadowing a Python builtin

(A002)